File: src/main/java/shuaicj/hobby/http/proxy/netty/HttpProxyChannelInitializer.java
package shuaicj.hobby.http.proxy.netty;

import io.netty.channel.ChannelInitializer;
import io.netty.channel.socket.SocketChannel;
import io.netty.handler.logging.LogLevel;
import io.netty.handler.logging.LoggingHandler;

public class HttpProxyChannelInitializer extends ChannelInitializer<SocketChannel> {

    @Override
    protected void initChannel(SocketChannel ch) {
        ch.pipeline().addLast(
                new LoggingHandler(LogLevel.DEBUG),
                new HttpProxyClientHandler()
        );
    }
}

File: src/main/scala/com/github/alikemalocalan/tunnel/HttpProxyChannelInitializer.scala
package com.github.alikemalocalan.tunnel

import io.netty.channel.ChannelInitializer
import io.netty.channel.socket.SocketChannel
import io.netty.handler.logging.{LogLevel, LoggingHandler}

class HttpProxyChannelInitializer extends ChannelInitializer[SocketChannel] {

  override def initChannel(ch: SocketChannel): Unit =
    ch.pipeline().addLast(
      new LoggingHandler(LogLevel.INFO),
      new HttpProxyClientHandler()
    )
}
